builtin/providers/aws/config.go

```diff
@@ -84,9 +86,15 @@ func (c *Config) Client() (interface{}, error) {
 			Region:      c.Region,
 			MaxRetries:  c.MaxRetries,
 		}
+		awsDynamoDBConfig := &aws.Config{
+			Credentials: creds,
+			Region:      c.Region,
+			MaxRetries:  c.MaxRetries,
+			Endpoint:    c.DynamoDBEndpoint,
+		}
```

Comment:
I didn't like this duplication of configs, but we can't set `Endpoint` for all other connections. I've also tried to copy `awsConfig`, but go didn't like it:

```go
awsDynamoDBConfig := * awsConfig
awsDynamoDBConfig.Endpoint = c.DynamoDBEndpoint
```

Any ideas?